### ISSUE TITLE
rpc: add ability to drop entire channel graph to re-sync

### ISSUE THREAD

**Issue by kroese** [lightningnetwork/lnd#6571]
When I run `describegraph` some JSON fields are empty, because of the setting `neutrino.validatechannels=false` that was temporarily activated by Umbrel while syncing the blockchain.

Is there any way to correct this without having to close all my channels and start from scratch with a clean database? 

It should technically be possible to overwrite the missing fields after Neutrino is disabled, so why does this not happen?

**Comment by Roasbeef** [comment]
The only way is to delete your channel graph and have it all re-synced. The channel graph is stored along with the other information, so it would need to be separated first. 

The only way to get all that information would be to 100% re-sync the entire graph (which also means fetching _all_ the relevant blocks to find out the true channel amount). As you add _new_ channels, it'll have the full capacity information.

**Comment by Roasbeef** [comment]
No, that stores other important information. You'll start to populate the information for _new_ channels. Switching between syncing backends isn't officially support, which can lead to behavior like what you're seeing.

**Comment by guggero** [comment]
You can use [`chantools dropchannelgraph`](https://github.com/guggero/chantools/blob/master/doc/chantools_dropchannelgraph.md) which does exactly what you need, just implemented in an external tool. It's not super well tested and not officially supported. So please use with caution until this is implemented in `lnd` itself.

**Comment by guggero** [comment]
The fee information for the own channels is also stored in the graph. That's why "resetting the graph" isn't a trivial task.
But I think you should be able to fix it by setting the fees for all your own channels. You should be able to do that for all channels at once with `lncli updatechanpolicy` if you _don't_ specify `--chan_point`.

**Comment by guggero** [comment]
I think you just need to let things run for a while and wait for all graph updates to come in. This shouldn't result in any force closures.

**Comment by guggero** [comment]
Thanks, I'm looking into it.

**Comment by guggero** [comment]
Okay, I know what the problem is. The own channels aren't added to the graph when they are announced on the network. So they need to be inserted after dropping the graph. I have added that functionality to `chantools`, so it should be relatively straight forward to fix. Can you contact me on [Slack](https://lightning.engineering/slack.html) please so I can instruct you on how to use the new functionality?

**Comment by dskvr** [comment]
@guggero Was this ever resolved? Had a power outage and could not resolve power outage before UPS ran out of power a couple weeks ago. Now it appears my graph is totally out of sync, creating rebalance issues, breaking `charge-lnd`'s ability to set fees (getchannelinfo returns `not found` for channels that should be found, for example) and also causing certain routing errors. The routing errors are huge as it is resulting in my channels not being checked for routing due to exorbitantly high failure rates (failure rate is 10x what is was before powoer outage). Fortunately, it appears my channel.db is fine which is a reliieve. Took ~2 weeks to figure out the problem. I agree with @kreoese that a graph resync should be native to `lnd`, as a healthy graph is absolutely essential to the functionality and extension of `lnd` and it's `rpc api`, the latter of which is the primary selling point of `lnd` over `cln`.

**Comment by guggero** [comment]
There is `chantools dropchannelgraph` but that should NOT be used as it doesn't correctly restore all local channels yet. I think some way of forcing a re-sync (instead of just dropping the whole graph) should be implemented in `lnd` itself...
Re-opening the issue with the hope that anyone has the time to look into it.
